packages/components/src/components/molecules/CardPile/CardPile.tsx

```diff
@@ -1,122 +1,152 @@
-    };
-    return !thisCard ? undefined : (
-      <div data-testid="CardPile_card" key={uuid()} className={classnames(classes.card, classes.firstItem)}>
-        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
-        <TagName {...cardProps} ref={firstCardRef} />
-      </div>
-    );
-  }, [thisCard, classes, component]);
-
-  const backFaceComponent = useMemo(() => {
-    const BackFaceTagName = backFace;
-    return pileCards.length <= 1 ? undefined : (
-      <div data-testid="CardPile-backface" className={classnames(classes.card, classes.lastItem)}>
-        <BackFaceTagName />
-      </div>
-    );
-  }, [pileCards.length, classes, backFace]);
+  const [isSelected, setSelect] = useState(false);
+  const [firstCard, secondCard, ...otherCards] = cards;
 
   const toggleSelect = () => {
     setSelect(!isSelected);
   };
-  const onPileClick = () => toggleSelect();
-  const onPileKeyDown = (event: KeyboardEvent<HTMLDivElement>) => event.keyCode === 32 && toggleSelect();
-  const onTurnFirstCard = () => firstCardRef.current?.flip();
-  const onShuffle = () => {
-    firstCardRef.current?.flip(false);
-    const newPile = shuffleArray(pileCards);
-    setPileCards(newPile);
+
+  const onTurnFirstCard = () => {
+    onCardFlipped(!isFaceUp);
+    setSelect(false);
+  };
+
+  const onShufflePile = () => {
+    onShuffle();
+    setSelect(false);
   };
 
   return (
-    <div
+    <Draggable
+      className={classes.cardPile}
+      position={position}
+      height={height}
+      width={width}
+      boundaries={boundaries}
+      onDragged={onPositionChanged}
+      onClick={toggleSelect}
+      // eslint-disable-next-line react/destructuring-assignment
       data-testid="CardPile"
-      role="button"
-      tabIndex={0}
-      onClick={onPileClick}
-      onKeyDown={onPileKeyDown}
-      className={classnames(classes.cardPile, { [classes.multiItem]: pileCards.length > 1 })}
     >
       {isSelected && (
         <>
           <Overlay />
           <div className={classes.cardPileMenu}>
-            <CardPileMenu onTurnFirstCard={onTurnFirstCard} onShufflePile={onShuffle} />
+            <CardPileMenu onTurnFirstCard={onTurnFirstCard} onShufflePile={onShufflePile} />
           </div>
         </>
       )}
-      {displayedCard}
-      {backFaceComponent}
-    </div>
+      {firstCard && (
+        <BaseCard
+          data-testid="CardPile-firstCard"
+          className={classes.firstItem}
+          width={width}
+          height={height}
+          frontFace={<PlayingCardFrontFace card={firstCard as PlayingCard} />}
+          backFace={<PlayingCardBackFace />}
+          disableNativeEvents
+          faceUp={isFaceUp}
+        />
+      )}
+      {secondCard && (
+        <BaseCard
+          data-testid="CardPile-secondCard"
+          className={classes.secondItem}
+          width={width}
+          height={height}
+          frontFace={<PlayingCardFrontFace card={secondCard as PlayingCard} />}
```

Comment:
What if it's not a PlayingCard? Like if it's a UnoCard
Also, do we need to render a full BaseCard (with front and back), given that you could only turn the first one?

Reply:
To be honest I wasn't sure about this change. I will probably change it back to the way it was so that it is more generic

Reply:
I added props `frontFace` and `backFace` to the component props so that it is more generic :)